stream.go

```diff
@@ -155,6 +160,11 @@ func (cs *clientStream) Trailer() metadata.MD {
 }
 
 func (cs *clientStream) SendMsg(m interface{}) (err error) {
+	cs.mu.Lock()
```

Comment:
An uncontended mutex lock/unlock is a compare-and-swap (for the lock) and an atomic add (for the unlock), which together is around 16ns on my (getting old) workstation. It's definitely not "hundreds of nanoseconds". It's only a little slower than reading a global var (EnableTracing), which may, in fact, be slower if there's lots of threads (a mutex per clientStream is thus effectively sharding that).

I suggest we keep this how this change has it. It keeps the code a bit simpler, it's an internal only thing, and we can switch it out when higher speed transports arrive and 16ns is enough to show up on a benchmark.